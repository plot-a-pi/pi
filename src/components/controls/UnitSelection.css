.UnitSelection {
  position: relative;
  bottom: 5.5vw;
  width: 16vw;
  z-index: 1;
  text-align: center;
  border-radius: .5rem;
  color: #212E59;
  padding: 1vw;

  input {
    width: 2.5vw;
    height: 2.5vw;
    padding:0;
    margin: 0;
  }

  label {
    font-size: 1.5vw;
    padding-right: 2vw;
  }
}