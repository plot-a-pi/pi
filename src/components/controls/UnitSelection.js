import React from 'react';
import { useDispatch } from 'react-redux';
import { changeUnit } from '../../actions/userActions';
import styles from './UnitSelection.css';
import { useSelector } from 'react-redux';
import { getUnit } from '../../selectors/userSelectors';

const UnitSelection = () => {

  const unit = useSelector(getUnit);
  const dispatch = useDispatch();

  return (
    <div className={styles.UnitSelection}>
      <label htmlFor='cm'><input id='cm' type='radio' name='unit' value='cm' checked={unit === 'cm'} onClick={() => dispatch(changeUnit('cm'))} />cm 
      </label>
      <label htmlFor='in'><input type='radio' name='unit' value='in' checked={unit === 'in'} onClick={() => dispatch(changeUnit('in'))} />in
      </label>
    </div>
  );
};

export default UnitSelection;
