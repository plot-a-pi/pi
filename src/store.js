const { createStore } = require('redux');
import userReducer from './reducers/userReducer';

export default createStore(
  userReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
