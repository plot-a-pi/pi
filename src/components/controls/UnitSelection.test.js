import React from 'react';
import { mount } from 'enzyme';
import UnitSelection from './UnitSelection';
import { Provider } from 'react-redux';
import store from '../../store';

describe('UnitSelection component', () => {
  it('renders the UnitSelection component', () => {
    const wrapper = mount(
      <Provider store={store} >
        <UnitSelection />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
