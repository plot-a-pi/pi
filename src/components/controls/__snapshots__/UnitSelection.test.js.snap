// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitSelection component renders the UnitSelection component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <UnitSelection>
    <div
      className="UnitSelection"
    >
      <input
        checked={true}
        id="cm"
        name="unit"
        onClick={[Function]}
        type="radio"
        value="cm"
      />
      <label
        htmlFor="cm"
      >
        cm
      </label>
      <input
        checked={false}
        name="unit"
        onClick={[Function]}
        type="radio"
        value="in"
      />
      <label
        htmlFor="in"
      >
        in
      </label>
    </div>
  </UnitSelection>
</Provider>
`;
